### ISSUE TITLE
TextField / Select wrong render when props changed fast

### ISSUE THREAD

**Issue by Wroud** [jamesmfriedman/rmwc#216]
### What RMWC Version are you using?

> checked on 1.4.1 and 1.5.5

### What browser(s) and React Version is this bug affecting?

> Google Chrome 66.0.3359.117, React 16.3.1

### What are the steps to reproduce the bug?

[![Edit RMWC Sandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/62jzl7l95w)  

> play with setTimeout delay & spam TEST button for reproducing bug

### Any other information you believe would be useful?

> Seems like bug depends of fields count and how fast render occurs
> Also if props `disabled=false` but rendered as `disabled=true` in react-devtools set prop to true
> `class` on div will be `mdc-text-field--upgraded mdc-text-field--disabled  mdc-text-field mdc-text-field--upgraded mdc-text-field--disabled`
>
> its like wrong class appending after props changed
> also after textfield rendered div has class="`mdc-text-field--upgraded  mdc-text-field mdc-text-field--upgraded`" why `mdc-text-field--upgraded` appended two times? its same bug?

**Comment by jamesmfriedman** [comment]
This is a really interesting bug. While React is responsible for rendering the markup, a lot of the work is delegated to the MDC code. There is definitely a communication issue somewhere in there.

Definitely something that I'll look into, but curious if it's blocking you right now, trying to figure out where to prioritize this one.

**Comment by Wroud** [comment]
checked code, look at [this line](https://github.com/jamesmfriedman/rmwc/blob/master/src/TextField/index.js#L187)
and [this one](https://github.com/jamesmfriedman/rmwc/blob/master/src/Base/MDCFoundation.js#L60)
```js
export const addClass = () =>
  function(className: string) {
    if (!this.state.classes.has(className)) {
      // The animation frame corrects an issue where MDC would set a class
      // on a form element and cause re-render before the new value could actually be set from the onChange
      setTimeout(() => {
        this._safeSetState(prevState => {
          if(prevState.classes.has(className)){
            return null;
          }
          return classes: prevState.classes.add(className)
        });
      });
    }
  };

export const removeClass = () =>
  function(className: string) {
    if (this.state.classes.has(className)) {
      // The animation frame corrects an issue where MDC would set a class
      // on a form element and cause re-render before the new value could actually be set from the onChange
      setTimeout(() => {
        this._safeSetState(prevState => {
          if(!this.state.classes.has(className)) {
            return null;
          }
          classes: prevState.classes.delete(className) ?
            prevState.classes :
            prevState.classes
        });
      });
    }
  };
```

**Comment by jamesmfriedman** [comment]
Thanks! Thats a huge help. I think I know a way around this, I'll look into it.

**Comment by rjdestigter** [comment]
@jamesmfriedman Is this maybe related? https://codesandbox.io/s/62040p3mw

Basically, for the TextField component, when the `value` prop changes to `undefined` it does not clear the input. I resolved this by sending in an empty string if the value is undefined or null but that renders the TextField in a state of invalidity even though it hasn't had focus yet.

**Comment by jamesmfriedman** [comment]
Fixed in 1.5.6, please verify in your codebases.

**Comment by Wroud** [comment]
It's seems fixed now, but i think you need move `if` statement in `._safeSetState((prevState)=> {here})` becose `this.state` and `prevState` can be different
